client/rest-high-level/src/main/java/org/elasticsearch/client/security/CreateApiKeyRequest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.client.security.user.privileges.Role;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Request to create API key
+ */
+public final class CreateApiKeyRequest implements Validatable, ToXContentObject {
+
+    private String name;
+    private TimeValue expiration;
+    private List<Role> roles = Collections.emptyList();
+    private final RefreshPolicy refreshPolicy;
+
+    /**
+     * Create API Key request constructor
+     * @param name name for the API key
+     * @param roles list of {@link Role}s
+     * @param expiration to specify expiration for the API key
+     */
+    public CreateApiKeyRequest(String name, List<Role> roles, @Nullable TimeValue expiration,
+            @Nullable final RefreshPolicy refreshPolicy) {
+        if (Strings.hasText(name)) {
+            this.name = name;
+        } else {
+            throw new IllegalArgumentException("name must not be null or empty");
+        }
+        this.roles = Objects.requireNonNull(roles, "roles may not be null");
+        this.expiration = expiration;
+        this.refreshPolicy = (refreshPolicy == null) ? RefreshPolicy.getDefault() : refreshPolicy;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
```

Comment:
why do we need setters? I'd prefer if we make all of the fields final and require them to be set in the constructor. Then we only have accessors